ts/WoltLabSuite/Core/Controller/User/Profile/TimezoneSync.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * Updates the user's timezone option on the server if the browser's
+ * timezone differs.
+ *
+ * @author  Tim Duesterhus
+ * @copyright   2001-2022 WoltLab GmbH
+ * @license GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @package WoltLabSuite\Core\Data\User
+ */
+
+import { getXsrfToken } from "../../../Core";
+
+function getServerTimezone(): string|undefined {
+  const meta = document.querySelector<HTMLMetaElement>('meta[name="timezone"]');
+
+  return meta?.content;
+}
+
+function getClientTimezone(): string
+{
+  return Intl.DateTimeFormat().resolvedOptions().timeZone
+}
+
+export async function maybeSyncServerTimezone(): Promise<void>
+{
+  if (getServerTimezone() !== getClientTimezone()) {
+    await fetch(`${window.WSC_API_URL}index.php?user-timezone-sync/`, {
```

Comment:
Fetch can fail loud. This should be enclosed in `try … catch`, because we do not care about failed requests (outside of the debug mode).